### ISSUE TITLE
Independent NuGet dependency declaration

### ISSUE THREAD

**Issue by andry-tino** [andry-tino/Rosetta#43]
Projects do not declare independent NuGet deps. Some projects need assemblies and make them as dependencies, however other projects just reference them withput having such deps stated in the `app.config`.

### Action items
- Detect every project having a dependency on NuGet packages (especially Roslyn)
- Make sure the cirresponding `app.config` exists in those projects.
- If not, create it and set deps there.

**Comment by michaelaird** [comment]
I believe this commit fixes this issue. as well as upgrading to the Roslyn 2.0:

https://github.com/michaelaird/Rosetta/commit/1d0297ebcbb6a6150a0cfea6662228de4f909b42

**Comment by andry-tino** [comment]
@michaelaird Yes, yep I just realized that I have a little mess in the deps and not completely handling them in the best way! Thanks

**Comment by michaelaird** [comment]
Let me know if you want to chat about integrating our changes back into the
main.

**Comment by andry-tino** [comment]
@michaelaird Yep definitely! So I can see the change comes after some changes you applied. We might be at a point with a lot of differences. Also I can see you have changed the .NET framework support to 4.6.

Did you add other projects up to Apr 8? If not I think the integration should succeed.

**Comment by michaelaird** [comment]
I've now merged all your changes into my fork so it should be pull-able.

I had to upgrade to .net 4.6 because the latest versions of the nuget packages for CodeAnalysis don't install properly on lower versions.

Let me know if you have any questions on what we've done.

**Comment by andry-tino** [comment]
HI Michael
Thanks and sorry for being late... These days have been quite something! Ok let's do this and try to merge all your changes into main here.

I am going to branch Rosetta main and then we do the merging. I have never done this before on GitHub so we'll see.
A few questions:

- Have you added new projects?
- Have you added tests?
- Have you tried running all the unit tests and renderers?

On Friday I will leave for vacation, I suggest we try this during this week :)
Also, let's continue this on #38 :)

**Comment by michaelaird** [comment]
I haven't added any new projects. 
I don't think I added any tests. :(
There is one test failing: WhenFileShortParameterSpecifiedThenFileRoutineIsExecuted

**Comment by andry-tino** [comment]
Yes, that is something that worries me a little. The code I have is pretty stable and I want to make sure it remains so. That test checks one of the runners. Chances are you broke the class structure and the test fails or something else.

At the moment I am actively working on #41 which is very important for my Team, so I cannot focus my energies on this integration at this moment. Looks like we need to go through the code together. So, let's wait for #41 to be done and then we should schedule a meeting together and have a look at your overall changes. I really want to integrate the changes so let's do this.

Also another question: which version of ScriptSharp are you targeting? Since you mentioned generics I see it must be one of the newest. I am asking because we must make sure to have Rosetta compatible with all versions of ScriptSharp.

**Comment by michaelaird** [comment]
No worries and no rush on my part. The translation units we've implemented
make Rosetta complete enough for translating our Script# projects so it's
been hugely helpful for us.

Feel free to pick and choose any changes you want to.

**Comment by andry-tino** [comment]
Thanks man. But those changes you pushed are gonna be valuable for my Team sooner or later so we need them :) I am gonna focus on integrating them after #41. With your help getting an overview of the changes you guys have pushed, I can also plan writing tests for those functionalities and stabilize them.

I will contact you soon! Thanks for your your efforts!
